cf/regrid/regrid.py

```diff
@@ -2000,3 +2390,110 @@ def update_non_coordinates(
 
         if axes and set(axes).issubset(dst_axis_keys):
             src.set_coordinate_reference(ref, parent=dst, strict=True)
+
+
+def update_data(src, regridded_data, src_grid):
+    """Insert the regridded field data.
+
+    .. versionadded:: 3.16.0
+
+    .. seealso: `update_coordinates`, `update_non_coordinates`
+
+    :Parameters:
+
+        src`: `Field`
+            The regridded field construct, that will up updated
+            in-place.
+
+        regridded_data: `numpy.ndarray`
+            The regridded array
+
+        src_grid: `Grid`
+            The definition of the source grid.
+
+    :Returns:
+
+        `None`
+
+    """
+    data_axes = src.get_data_axes()
+    if src_grid.new_axis_keys:
+        # The regridding has changed the number of data axes (e.g. by
+        # regridding a source UGRID grid to a destination non-UGRID
+        # grid, or vice versa) => delete the old, superceded domain
+        # axis construct and update the list of data axes.
+        data_axes = list(data_axes)
+        index = data_axes.index(src_grid.axis_keys[0])
+        for axis in src_grid.axis_keys:
+            data_axes.remove(axis)
+            src.del_construct(axis)
+
+        data_axes[index:index] = src_grid.new_axis_keys
+
+    src.set_data(regridded_data, axes=data_axes, copy=False)
+
+
+def get_mesh(f):
+    """Get domain topology mesh information.
+
+    .. versionadded:: 3.16.0
+
+    :Parameters:
+
+        f: `Field` or `Domain`
+            The construct from which to get the mesh information.
+
+    :Returns:
+
+       3-`tuple`
+           If the field or domain has no domain topology construct
+           then ``(None, None, None)`` is returned. Otherwise the
+           tuple contains:
+
+           * The domain topology construct
+           * The mesh location of the domain topology (e.g. ``'face'``)
+           * The identifier of domain axis construct that is spanned
+             by the domain topology construct
+
+    """
+    key, domain_topology = f.domain_topology(item=True, default=(None, None))
+    if domain_topology is None:
+        return (None, None, None)
+
+    return (
+        domain_topology,
+        domain_topology.get_cell(""),
+        f.get_data_axes(key)[0],
+    )
+
+
+def has_coordinate_arrays(grid):
+    """Whether all grid coordinates have representative arrays.
+
+    .. versionadded:: 3.16.0
+
+    :Parameters:
+
+        grid: `Grid`
+            The definition of the grid.
+
+    :Returns:
+
+        `bool`
+            True if and only if all grid coordinates have
+            representative arrays.
```

Comment:
```suggestion
            True if and only if there are grid coordinates and they
            all have representative arrays.
```

or something else to clarify the output when there are no grid coordinates, would be ideal to cover all possibilities.